File: Andriod/Button/app/src/main/java/com/example/button/MainActivity.java
package com.example.button;

import androidx.appcompat.app.AppCompatActivity;

import android.app.DatePickerDialog;
import android.app.TimePickerDialog;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.DatePicker;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.TimePicker;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.List;

import butterknife.BindDimen;
import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{

    @BindView(R.id.toolbar)
    View toolbar;
    ImageView ivBack;
    TextView tvHeading;

    @BindView(R.id.rlDatePicker)
    RelativeLayout rlDatePicker;
    @BindView(R.id.lltimePicker)
    RelativeLayout lltimePicker;
    @BindView(R.id.rlSpinnerPicker)
    RelativeLayout rlSpinnerPicker;
    @BindView(R.id.tvDate)
    TextView tvDate;
    @BindView(R.id.tvTime)
    TextView tvTime;
    @BindView(R.id.spinnerText)
    Spinner spinnerText;
    public String name[]={"India","Australia","South Africa","West Indies","New Zealand","Sri Lanka","Bangladesh"};
    private List<String> list;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        rlDatePicker.setOnClickListener(this);
        lltimePicker.setOnClickListener(this);
        settoolbar();

        list=new ArrayList<String>();
        for(int i=0;i<name.length;i++){
            list.add(name[i]);
        }
        ArrayAdapter<String> adapter=new ArrayAdapter<String>(this,android.R.layout.simple_spinner_dropdown_item,list);
        spinnerText.setAdapter(adapter);


    }

    public void settoolbar()
    {
        ivBack=toolbar.findViewById(R.id.ivBack);
        tvHeading=toolbar.findViewById(R.id.tvHeading);
        tvHeading.setText("Demo Pratice");

    }

    @Override
    public void onClick(View view) {
        switch(view.getId()){
            case R.id.rlDatePicker:
                openDatePicker();
                break;
            case R.id.lltimePicker:
                openTimePicker();
                break;
            default:
                break;
        }
    }

    public void openDatePicker()
    {
        Calendar mcurrentDate=Calendar.getInstance();
        int year=mcurrentDate.get(Calendar.YEAR);
        int month=mcurrentDate.get(Calendar.MONTH);
        int day=mcurrentDate.get(Calendar.DAY_OF_MONTH);
        Date time=mcurrentDate.getTime();

        final DatePickerDialog   mDatePicker =new DatePickerDialog(MainActivity.this, new DatePickerDialog.OnDateSetListener()
        {
            @Override
            public void onDateSet(DatePicker datepicker, int selectedyear, int selectedmonth, int selectedday)
            {
                tvDate.setText(new StringBuilder().append(selectedday).append("/").append(selectedmonth+1).append("/").append(selectedyear));
                Toast.makeText(getApplicationContext(),String.valueOf(time),Toast.LENGTH_SHORT).show();
            }
        },year, month, day);
        mDatePicker.getDatePicker().setMaxDate(System.currentTimeMillis());
        mDatePicker.show();
    }

    public void openTimePicker(){

        Calendar calendar=Calendar.getInstance();
        int hour=calendar.get(Calendar.HOUR_OF_DAY);
        int minute=calendar.get(Calendar.MINUTE);
        TimePickerDialog timePicker=new TimePickerDialog(MainActivity.this, new TimePickerDialog.OnTimeSetListener() {
            @Override
            public void onTimeSet(TimePicker timePicker, int hour, int minutes) {
                tvTime.setText(String.valueOf(hour)+":"+String.valueOf(minutes));
            }
        },hour,minute,true);
        timePicker.show();
    }

}

File: Andriod/DeviceStorage/app/src/main/java/com/traversal/devicestorage/MainActivity.kt
package com.traversal.devicestorage

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Environment
import android.os.StatFs
import android.view.LayoutInflater
import android.view.View
import android.widget.Toast
import com.traversal.devicestorage.databinding.ActivityMainBinding
import java.io.File
import android.system.Os.stat




class MainActivity : AppCompatActivity() , View.OnClickListener {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(LayoutInflater.from(this))
        setContentView(binding.root)
        binding.btnStorage.setOnClickListener(this)
    }

    private fun deviceStorage() {
        if(isExternalMemoryAvailable()){
            val size = totalExternalMemorySize()
            Toast.makeText(this,size,Toast.LENGTH_SHORT).show()
        }else{
            Toast.makeText(this,"External Memory Not Available",Toast.LENGTH_SHORT).show()
        }
    }

    override fun onClick(view: View?) {
        when(view?.id){
           R.id.btnStorage -> {
               deviceStorage()
           }
           else -> {
               Toast.makeText(this,"Click Appropriate Button",Toast.LENGTH_SHORT).show()
           }
        }
    }

    private fun totalExternalMemorySize() : String {
        val path = Environment.getDataDirectory()
        val stat = StatFs(path.path)
        val blockSize = stat.blockSizeLong
        val availableBlocks = stat.availableBlocksLong
        return (availableBlocks * blockSize).toString()
    }

    private fun isExternalMemoryAvailable() : Boolean {
        return Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)
    }
}